src/wp-includes/functions.php

```diff
@@ -2479,14 +2479,22 @@ function _wp_upload_dir( $time = null ) {
 	$baseurl = $url;
 
 	$subdir = '';
-	if ( get_option( 'uploads_use_yearmonth_folders' ) ) {
+	$storefolders = get_option( 'uploads_use_yearmonth_folders' );
+	if ( $storefolders === '1' ) {
 		// Generate the yearly and monthly directories.
 		if ( ! $time ) {
 			$time = current_time( 'mysql' );
 		}
 		$y      = substr( $time, 0, 4 );
 		$m      = substr( $time, 5, 2 );
 		$subdir = "/$y/$m";
+	} elseif ( $storefolders === 'year' ) {
```

Comment:
As above, I think it would be more consistent if the option value for the new structure was '2'.